### DISCUSSION TITLE
auto increment on a non-pk column

### DISCUSSION THREAD

**Discussion by liuzqt** [sqlalchemy/sqlalchemy#6599] [Usage Questions] [answered]
**Describe the use case**
I want to make a non-pk column auto increment

**Databases / Backends / Drivers targeted**
sqlalchemy version: 1.3

postgressql
psycopg2

**Example Use**
In psql I can do this:
```
CREATE TABLE map_objects (
        oid BIGSERIAL NOT NULL, 
        vid BIGSERIAL NOT NULL, 
        msg VARCHAR, 
        tags VARCHAR[], 
        PRIMARY KEY (vid)
);


```

Just want to have something similar in sqlalchemy. I've done some search on stackoverflow, seems that this feature is not supported in 1.1,  not sure if we already have this in latest version. I've looked through the document but didn't find the answer.

Here is my code defining the model:

```
class MapObject(ModelBase):
    __tablename__ = 'map_objects'
    oid = Column(BigInteger, autoincrement=True)
    vid = Column(BigInteger, primary_key=True, autoincrement=True)
    msg = Column(String, nullable=True)
    tags = Column(POSTGRES_ARRAY(String))
```

however, the create table statement created by sqlalchemy is:
```
CREATE TABLE map_objects (
        oid BIGINT, 
        vid BIGSERIAL NOT NULL, 
        msg VARCHAR, 
        tags VARCHAR[], 
        PRIMARY KEY (vid)
)
```

noticed that the auto increment attribute in `oid` column is ignored.

**Have a nice day!**

**Comment by CaselIT** [comment, marked as answer]
Hi,

As mentioned in the docs https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sequences-serial-identity you can use an identity column (or a sequence directly) to specify an autoincement column.

**Reply by zzzeek** [reply]
side node, they added labels to github discussions ?  wow

**Reply by zzzeek** [reply]
can a PG table have multiple IDENTITY columns?

**Reply by CaselIT** [reply]
yes, identity is basically the same as a sequence in pg, so yes, you can have as many as you want

> side node, they added labels to github discussions ? wow

oh cool

**Reply by liuzqt** [reply]
@CaselIT  thanks for your answer, it solved my problem perfectly!
